lib/ash_json_api/json_schema/open_api.ex

```diff
@@ -0,0 +1,956 @@
+                type: :string
+              },
+              parameter: %Schema{
+                description: "A string indicating which query parameter caused the error.",
+                type: :string
+              }
+            }
+          }
+          # "meta" => %{
+          #   "$ref" => "#/definitions/meta"
+          # }
+        },
+        additionalProperties: false
+      }
+    }
+  end
+
+  defp resources(api) do
+    api
+    |> Ash.Api.Info.resources()
+    |> Enum.filter(&(AshJsonApi.Resource in Spark.extensions(&1)))
+  end
+
+  @spec resource_object_schema(resource :: module) :: Schema.t()
+  defp resource_object_schema(resource) do
+    %Schema{
+      description:
+        "A \"Resource object\" representing a #{AshJsonApi.Resource.Info.type(resource)}",
+      type: :object,
+      required: [:type, :id],
+      properties: %{
+        type: %Schema{type: :string},
+        id: %{type: :string},
+        attributes: attributes(resource),
+        relationships: relationships(resource)
+        # "meta" => %{
+        #   "$ref" => "#/definitions/meta"
+        # }
+      },
+      additionalProperties: false
+    }
+  end
+
+  @spec attributes(resource :: module) :: Schema.t()
+  defp attributes(resource) do
+    %Schema{
+      description: "An attributes object for a #{AshJsonApi.Resource.Info.type(resource)}",
+      type: :object,
+      properties: resource_attributes(resource),
+      additionalProperties: false,
+      required: required_attributes(resource)
+    }
+  end
+
+  @spec resource_attributes(resource :: module) :: %{atom => Schema.t()}
+  defp resource_attributes(resource) do
+    resource
+    |> Ash.Resource.Info.public_attributes()
+    |> Map.new(fn attr ->
+      {attr.name, resource_attribute_type(attr)}
+    end)
+  end
+
+  @spec resource_attribute_type(%{type: module}) :: Schema.t()
+  defp resource_attribute_type(%{type: Ash.Type.String}) do
+    %Schema{type: :string}
+  end
+
+  defp resource_attribute_type(%{type: Ash.Type.Boolean}) do
+    %Schema{type: :boolean}
+  end
+
+  defp resource_attribute_type(%{type: Ash.Type.Integer}) do
+    %Schema{type: :integer}
+  end
+
+  defp resource_attribute_type(%{type: Ash.Type.UtcDatetime}) do
+    %Schema{
+      type: :string,
+      format: "date-time"
+    }
+  end
+
+  defp resource_attribute_type(%{type: Ash.Type.UUID}) do
+    %Schema{
+      type: :string,
+      format: "uuid"
+    }
+  end
+
+  defp resource_attribute_type(%{type: {:array, type}}) do
+    %Schema{
+      type: :array,
+      items: resource_attribute_type(%{type: type})
+    }
+  end
+
+  defp resource_attribute_type(%{type: type} = attr) do
+    if :erlang.function_exported(type, :json_schema, 1) do
+      if Map.get(attr, :constraints) do
+        type.json_schema(attr.constraints)
+      else
+        type.json_schema([])
+      end
+    else
+      %Schema{
+        type: :object,
+        additionalProperties: true
```

Comment:
Should we fall back to `string` here instead of an object for types we can't determine?